drivers/scheduler/k8s/k8s.go

```diff
@@ -616,6 +616,72 @@ func (k *k8s) GetNodesForApp(ctx *scheduler.Context) ([]node.Node, error) {
 	return result, nil
 }
 
+func (k *k8s) GenerateSupportBundle(ctx *scheduler.Context) (string, error) {
+	var output string
+	var err error
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			output += insertLineBreak(obj.Name)
+			var depStatus apps_api.DeploymentStatus
+			if depStatus, err = k8sutils.DescribeDeployment(obj); err != nil {
+				output += fmt.Sprintf("%v", &ErrFailedToValidateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get status of deployment: %v. Err: %v", obj.Name, err),
+				})
+			}
+			//Dump depStatus
+			output += fmt.Sprintf("%v\n", depStatus)
+			pods, _ := k8sutils.GetDeploymentPods(obj)
+			for _, pod := range pods {
+				output += insertLineBreak(pod.Name)
+				output += fmt.Sprintf("%v\n", pod.Status)
+				for _, conStat := range pod.Status.ContainerStatuses {
+					output += insertLineBreak(conStat.Name)
+					output += fmt.Sprintf("%v\n", conStat)
+					output += insertLineBreak("END Container")
+				}
+				output += insertLineBreak("END Pod")
+			}
+			output += insertLineBreak("END Deployment")
+		} else if obj, ok := spec.(*apps_api.StatefulSet); ok {
+			output += insertLineBreak(obj.Name)
+			var ssetStatus apps_api.StatefulSetStatus
+			if ssetStatus, err = k8sutils.DescribeStatefulSet(obj); err != nil {
+				output += fmt.Sprintf("%v", &ErrFailedToValidateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get status of statefulset: %v. Err: %v", obj.Name, err),
+				})
+			}
+			//Dump ssetStatus
+			output += fmt.Sprintf("%v\n", ssetStatus)
+			pods, _ := k8sutils.GetStatefulSetPods(obj)
+			for _, pod := range pods {
+				output += insertLineBreak(pod.Name)
+				output += fmt.Sprintf("%v\n", pod.Status)
+				for _, conStat := range pod.Status.ContainerStatuses {
+					output += insertLineBreak(conStat.Name)
+					output += fmt.Sprintf("%v\n", conStat)
+					output += insertLineBreak("END Container")
+				}
+				output += insertLineBreak("END Pod")
+			}
+			output += insertLineBreak("END StatefulSet")
+		} else {
+			output += insertLineBreak("UNKNOWN/UNSUPPORTED OBJ TYPE")
+			output += fmt.Sprintf("%v\n", &ErrFailedToValidateApp{
+				App:   ctx.App,
+				Cause: fmt.Sprintf("Unsupported app type: %v", obj),
+			})
+			output += insertLineBreak("END UNSUPPORTED OBJ TYPE")
+		}
```

Comment:
We need to dump PVC and Service status too.

Reply:
@harsh-px my original intent as I explained to @piyush-nimbalkar was to leave a log in place that points out an unsupported object and so we can identify how to add support to that. Currently PVC and Service are two such objects but when I did started this branch we hasn't yet thought of these objects.
To dump the status of these as well we will have to add underlying status calls. I can do that but I hadn't accounted for that for this effort. I'll work on adding those too.